cmake/onnxruntime_java.cmake

```diff
@@ -62,49 +59,15 @@ file(GLOB onnxruntime4j_native_src
     "${REPO_ROOT}/include/onnxruntime/core/session/*.h"
     )
 # Build the JNI library
-add_library(onnxruntime4j_jni SHARED ${onnxruntime4j_native_src})
+onnxruntime_add_shared_library_module(onnxruntime4j_jni ${onnxruntime4j_native_src})
 set_property(TARGET onnxruntime4j_jni PROPERTY CXX_STANDARD 11)
 
-# Tell the JNI code about the requested providers
```

Comment:
Why is this code removed? It's used to cause the JNI code to emit a well formed exception if the provider isn't available.

Reply:
I moved it to CMakeLists.txt.  And I think the things like "USE_DNNL" should not be used in the JNI project. I suggest the macro should be always on.

Reply:
Now that the provider list exists in the C API I guess I can check that and throw the exception, but if you look in SessionOptions.c it's used to either compile the JNI method so it always throws a Java exception, or to compile it so it adds the requested provider. I'm not sure how stable that provider list is though, as it's not in the C API itself, I've got to manually copy the string constants into Java and check them.